### ISSUE TITLE
Mapping page is blank for extensions to Taxon core

### ISSUE THREAD

**Issue by peterdesmet** [gbif/ipt#1204]
I've just upgraded our IPT to 2.3.1. One of our datasets is a checklist, but I'm now getting blank mapping pages for the extensions SpeciesDistribution and VernacularNames:

1. Remove all mapping
2. Add mapping to Taxon core: works
3. Add mapping to SpeciesDistribution: a blank page is shown, leaving no possibility to edit the mapping. It does indicate that the **automapping** worked though:

    ![screen shot 2015-10-02 at 14 23 17](https://cloud.githubusercontent.com/assets/600993/10246476/0ce8418a-6912-11e5-9bbb-4791bc204dda.png)

4. Click save. An error page appears with: `We are sorry, but the requested resource does not exist.`
5. Navigate back to the resource management page: the mapping seemed to have worked and can be previewed:

    ![screen shot 2015-10-02 at 14 32 51](https://cloud.githubusercontent.com/assets/600993/10246527/83727802-6912-11e5-97ba-d3c204705849.png)

6. Idem for vernacularNames

So, the mapping functionality is not broken, but the UI is.

Note: I'm using the same source file for core and extensions, as it is a one-to-one mapping between all.

**Comment by timrobertson100** [comment]
This is the freemarker error shown:

```
INFO  [org.gbif.ipt.action.manage.MappingAction] - Core row type: http://rs.tdwg.org/dwc/terms/Taxon
INFO  [org.gbif.ipt.action.manage.MappingAction] - Field representing the id for the core: http://rs.tdwg.org/dwc/terms/taxonID
ERROR [freemarker.runtime] - Error executing FreeMarker template
FreeMarker template error:
The following has evaluated to null or missing:
==> mapping.extension.link  [in template "WEB-INF/pages/manage/mapping.ftl" at line 330, column 294]

----
Tip: It's the step after the last dot that caused this error, not those before it.
----
Tip: If the failing expression is known to be legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use <#if myOptionalVar??>when-present<#else>when-missing</#if>. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
----

----
FTL stack trace ("~" means nesting-related):
	- Failed at: ${mapping.extension.link} auto-escaped  [in template "WEB-INF/pages/manage/mapping.ftl" at line 330, column 292]
----

Java stack trace (for programmers):
----
freemarker.core.InvalidReferenceException: [... Exception message was already printed; see it above ...]
	at freemarker.core.InvalidReferenceException.getInstance(InvalidReferenceException.java:131)
	at freemarker.core.EvalUtil.coerceModelToString(EvalUtil.java:355)
	at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
	at freemarker.core.BuiltInForString._eval(BuiltInForString.java:26)
	at freemarker.core.Expression.eval(Expression.java:78)
	at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
	at freemarker.core.DollarVariable.accept(DollarVariable.java:41)
	at freemarker.core.Environment.visit(Environment.java:324)
	at freemarker.core.MixedContent.accept(MixedContent.java:54)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:345)
	at freemarker.core.Environment.visitAndTransform(Environment.java:425)
	at freemarker.core.UnifiedCall.accept(UnifiedCall.java:107)
	at freemarker.core.Environment.visit(Environment.java:324)
	at freemarker.core.MixedContent.accept(MixedContent.java:54)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:345)
	at freemarker.core.Environment.visitAndTransform(Environment.java:425)
	at freemarker.core.UnifiedCall.accept(UnifiedCall.java:107)
	at freemarker.core.Environment.visit(Environment.java:324)
	at freemarker.core.MixedContent.accept(MixedContent.java:54)
	at freemarker.core.Environment.visit(Environment.java:324)
	at freemarker.core.EscapeBlock.accept(EscapeBlock.java:48)
	at freemarker.core.Environment.visit(Environment.java:324)
	at freemarker.core.MixedContent.accept(MixedContent.java:54)
	at freemarker.core.Environment.visit(Environment.java:324)
	at freemarker.core.Environment.process(Environment.java:302)
	at freemarker.template.Template.process(Template.java:325)
```

**Comment by timrobertson100** [comment]
This is caused because the extensions do not have a link:
  http://rs.gbif.org/extension/gbif/1.0/speciesprofile.xml

And this line [1] in the code has very tight freemarker which fails if it is missing.  It has:
```
  <a href="${mapping.extension.link}">
```

The code should be relaxed to accommodate that, but a simple quick fix would be to add links to the extensions.

[1] https://github.com/gbif/ipt/blob/master/src/main/webapp/WEB-INF/pages/manage/mapping.ftl#L330

**Comment by kbraak** [comment]
Thanks for debugging this issue Tim.

Links have now been added to the extensions. 

Affected users (@peterdesmet) can fix the problem by 1) pressing the “synchronise” button (on the Admin/Core Types and Extensions page) and 2) restarting the servlet container. 

After the code has been relaxed as mentioned above, this issue can be closed.

**Comment by peterdesmet** [comment]
Confirmed fixed in `2.3.2 SNAPSHOT`. I did NOT click synchronize or restart servlet container.
